### PR TITLE
Bump versions and update dev scripts

### PR DESCRIPTION
**Pull request by dbernheisel**
- Bump Elixir to 1.6.1
  - Fix warnings by removing trailing commas in macro functions
  - Adjust Heroku buildpack to match
- Bump Erlang to 20.2.3
- Specify NodeJS version 9.2.0
  - Add `package-lock.json`
  - Have CircleCI install same version of NodeJS as in the dev environment
  - Have CircleCI cache NPM modules based on package-lock.
  - Specify some NPM dependencies as dev dependencies.
- Bump all Elixir dependencies and adjust for upgrades
- Update local dev scripts
  - Move PhantomJS runner to start in `test_helper.exs` instead of `bin/test_suite`. This lets us run `mix test` and have feature tests work.
  - Make seeds faster by truncating tables instead of dropping/creating
  - Add `bin/server` which resets database, re-seeds, and starts web server
  - Adjust username for test database to not require "POSTGRES" user. (now it matches dev environment behavior)
- Add Phoenix buildpack for Heroku.

### CODE REVIEW

File: bin/setup [hunk 1]
```diff
@@ -28,6 +28,16 @@ if [ -z "$CI" ]; then
   rm -rf node_modules
   rm -rf deps _build
   asdf install
+else
+  [[ ! -d ~/.asdf ]] && git clone https://github.com/asdf-vm/asdf.git ~/.asdf --branch v0.4.1
+  . $HOME/.asdf/asdf.sh
+  echo -e '\n. $HOME/.asdf/asdf.sh' >> ~/.bashrc
+  if ! grep -E nodejs <(asdf plugin-list) > /dev/null; then
+    asdf plugin-add nodejs https://github.com/asdf-vm/asdf-nodejs.git
+    bash ~/.asdf/plugins/nodejs/bin/import-release-team-keyring
```
Comment (drapergeek) [inline bin/setup:37]: I don't think we should do this in a setup script. This is adding GPG keys to someone's keyring which doesn't feel like the kind of thing we should do automatically, especially in a public project.

Maybe we should put out that they need to do this?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,53 +1,64 @@
...
 with the necessary dependencies to run and test this app:
 
     % ./bin/setup
 
-It assumes you have the following:
-
-* Postgresql
-* Elixir
-* PhantomJS
+It assumes you have a machine equipped with [asdf], [asdf-elixir], [asdf-erlang],
```
Comment (drapergeek) [inline CONTRIBUTING.md:19]: We don't actually assume the plugins, we only assume asdf now that the script is updated, correct?

**Review by drapergeek — APPROVED** [review]

File: bin/setup (hunk 1 above)
Comment (dbernheisel) [reply bin/setup:37]: Thanks. Ya, that's not cool. I was thinking in the interest of getting the project ready in a single command, but you're right, I wouldn't want some random project telling my computer to trust others' signatures.

File: CONTRIBUTING.md (hunk 2 above)
Comment (dbernheisel) [reply CONTRIBUTING.md:19]: To run `bin/setup` we do assume the plugins are installed, otherwise `asdf install` will fail.

File: bin/setup (hunk 1 above)
Comment (dbernheisel) [reply bin/setup:37]: On second thought, this section of code is about handling CI setup. In that case, I don't think this matters. We only run `asdf install` on the person's computer, which might fail for nodejs if they don't follow asdf-nodejs's instructions about gpg signatures and all that.

Comment (drapergeek) [reply bin/setup:37]: Ah, sorry I misread that. Rock on.

File: CONTRIBUTING.md (hunk 2 above)
Comment (drapergeek) [reply CONTRIBUTING.md:19]: Cool.

**Pull request merged**